src/smvlang/smv_typecheck.cpp

```diff
@@ -1911,43 +1904,56 @@ Function: smv_typecheckt::convert_vars
 
 \*******************************************************************/
 
-void smv_typecheckt::convert(smv_parse_treet::mc_varst &vars)
+void smv_typecheckt::create_var_symbols(
+  const smv_parse_treet::modulet::item_listt &items)
 {
   symbolt symbol;
 
   symbol.mode="SMV";
   symbol.module=modulep->name;
 
-  for(const auto &var_it : vars)
+  for(const auto &item : items)
   {
-    const smv_parse_treet::mc_vart &var = var_it.second;
-
-    typet type = var.type;
+    if(item.is_var())
+    {
+      typet type = item.expr.type();
 
-    // check the type, if any
-    if(type.is_not_nil())
+      // check the type
       check_type(type);
 
-    symbol.base_name = var_it.first;
+      symbol.base_name = to_symbol_expr(item.expr).get_identifier();
```

Comment:
This re-use of the same symbolt storage makes me a bit uncomfortable. How about creating a fresh `symbolt` (then using the most suitable constructor) in here and then using `symbol_table.insert(std::move(symbol));` instead of `symbol_table.add`?

Reply:
ok, done